packages/react-mutation-mapper/src/component/oncokb/OncoKbCardTreatmentContent.tsx

```diff
@@ -0,0 +1,280 @@
+                                style={{
+                                    maxHeight: 200,
+                                    overflowY: 'auto',
+                                }}
+                            >
+                                {this.props.biologicalSummary !== undefined &&
+                                this.props.biologicalSummary.length > 0 ? (
+                                    <SummaryWithRefs
+                                        content={this.props.biologicalSummary}
+                                        type={'tooltip'}
+                                        pmidData={this.props.pmidData!}
+                                    />
+                                ) : this.props.mutationEffectCitations &&
+                                  (this.props.mutationEffectCitations.abstracts
+                                      .length > 0 ||
+                                      this.props.mutationEffectCitations.pmids
+                                          .length > 0) ? (
+                                    <ReferenceList
+                                        pmidData={this.props.pmidData}
+                                        pmids={this.props.mutationEffectCitations!.pmids.map(
+                                            pmid => Number(pmid)
+                                        )}
+                                        abstracts={
+                                            this.props.mutationEffectCitations!
+                                                .abstracts
+                                        }
+                                    />
+                                ) : (
+                                    <span>
+                                        Mutation effect information is not
+                                        available.
+                                    </span>
+                                )}
+                            </div>
+                        )}
+                    </div>
+                </span>
+            </div>
+        );
+    }
+
+    @autobind
+    private handleOncogenicityTabSelect(): void {
+        this.handleTabSelect(ActiveTabEnum.ONCOGENICITY);
+    }
+
+    @autobind
+    private handleMutationEffectTabSelect(): void {
+        this.handleTabSelect(ActiveTabEnum.MUTATION_EFFECT);
+    }
+
+    @autobind
+    @action
+    handleTabSelect(tabName: ActiveTabEnum): void {
+        this.activeTab = tabName;
+    }
+
+    public insertLink(str: string, link: any) {
+        if (!str) {
+            return str;
+        }
+
+        const content: Array<string | JSX.Element> = [];
+        const parts = str.split(link.keyword);
+
+        content.push(parts[0]);
+
+        const comp = (
+            <a href={link.link} target={link.target || '_blank'}>
+                {link.keyword}
+            </a>
+        );
+
+        for (let i = 1; i < parts.length; i++) {
```

Comment:
parts.forEach. or even   parts reduce(..., content).   i just don't like plain for loops unless the concern is performance.